crates/viewer/re_dataframe_ui/src/filters/string.rs

```diff
@@ -0,0 +1,299 @@
+        match self {
+            Self::Contains => "contains".fmt(f),
+            Self::StartsWith => "starts with".fmt(f),
+            Self::EndsWith => "ends with".fmt(f),
+        }
+    }
+}
+
+impl StringOperator {
+    pub const ALL: &'static [Self] = &[Self::Contains, Self::StartsWith, Self::EndsWith];
+}
+
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct StringFilter {
+    operator: StringOperator,
+    query: String,
+}
+
+impl SyntaxHighlighting for StringFilter {
+    fn syntax_highlight_into(&self, builder: &mut SyntaxHighlightedBuilder) {
+        builder.append_string_value(&self.query);
+    }
+}
+
+impl StringFilter {
+    pub fn new(operator: StringOperator, query: impl Into<String>) -> Self {
+        Self {
+            operator,
+            query: query.into(),
+        }
+    }
+
+    pub fn operator(&self) -> StringOperator {
+        self.operator
+    }
+
+    pub fn as_filter_expression(&self, column: &Column) -> Expr {
+        if self.query.is_empty() {
+            return lit(true);
+        }
+
+        let udf = ScalarUDF::new_from_impl(StringFilterUdf::new(self));
+        udf.call(vec![col(column.clone())])
+    }
+
+    pub fn popup_ui(
+        &mut self,
+        ui: &mut egui::Ui,
+        column_name: &str,
+        popup_just_opened: bool,
+    ) -> FilterUiAction {
+        let operator_text = self.operator.to_string();
+
+        ui.horizontal(|ui| {
+            ui.label(
+                SyntaxHighlightedBuilder::body_default(column_name).into_widget_text(ui.style()),
+            );
+
+            egui::ComboBox::new("string_op", "")
+                .selected_text(
+                    SyntaxHighlightedBuilder::keyword(&operator_text).into_widget_text(ui.style()),
+                )
+                .show_ui(ui, |ui| {
+                    for possible_op in crate::filters::StringOperator::ALL {
+                        if ui
+                            .button(
+                                SyntaxHighlightedBuilder::keyword(&possible_op.to_string())
+                                    .into_widget_text(ui.style()),
+                            )
+                            .clicked()
+                        {
+                            self.operator = *possible_op;
+                        }
+                    }
+                });
+        });
+
+        let response = ui.text_edit_singleline(&mut self.query);
+
+        if popup_just_opened {
+            response.request_focus();
+        }
+
+        action_from_text_edit_response(ui, &response)
+    }
+}
+
+pub fn is_supported_string_datatype(data_type: &DataType) -> bool {
+    matches!(
+        data_type,
+        DataType::Utf8 | DataType::Utf8View | DataType::LargeUtf8
+    )
+}
+
+/// Custom UDF for performing filtering on a column of strings, with support for list columns.
+///
+/// This UDF converts both the haystack and the needle to lowercase before performing the queries.
```

Comment:
we probably should put the case insensitiveness on a tooltip somewhere? might get cluttery though tbf

Reply:
It could be a note below the text entry field. All ask Katya what she thinks about that.